test-network/organizations/cfssl/registerEnroll.sh

```diff
@@ -0,0 +1,276 @@
+#!/bin/bash
+# Copyright 2023 Aditya Joshi, All rights reserved
+
+function peer_cert() {
+
+    TYPE=$1 #peer user
+    USER=$2
+    ORG=$3
+
+    mkdir -p "organizations/peerOrganizations/$ORG.example.com/ca"
+    mkdir -p "organizations/peerOrganizations/$ORG.example.com/msp/cacerts"
+    mkdir -p "organizations/peerOrganizations/$ORG.example.com/msp/tlscacerts"
+    mkdir -p "organizations/peerOrganizations/$ORG.example.com/peers"
+    mkdir -p "organizations/peerOrganizations/$ORG.example.com/tlsca"
+
+    CERT_DIR=organizations/peerOrganizations/$ORG.example.com
+
+    if [ ! -f "$CERT_DIR/ca/ca-key.pem" ]; then
+
+        cfssl gencert -initca "${PWD}/organizations/cfssl/ca-peer.json" | cfssljson -bare "$CERT_DIR/ca/ca"
+
+        cp "$CERT_DIR/ca/ca.pem" "$CERT_DIR/tlsca/tlsca.$ORG.example.com-cert.pem"
+        cp "$CERT_DIR/ca/ca.pem" "$CERT_DIR/ca/ca.$ORG.example.com-cert.pem"
+
+        cp "$CERT_DIR/ca/ca.pem" "$CERT_DIR/msp/cacerts/"
+        cp "$CERT_DIR/ca/ca.pem" "$CERT_DIR/msp/tlscacerts/"
+
+        echo 'NodeOUs:
+    Enable: true
+    ClientOUIdentifier:
+        Certificate: cacerts/ca.pem
+        OrganizationalUnitIdentifier: client
+    PeerOUIdentifier:
+        Certificate: cacerts/ca.pem
+        OrganizationalUnitIdentifier: peer
+    AdminOUIdentifier:
+        Certificate: cacerts/ca.pem
+        OrganizationalUnitIdentifier: admin
+    OrdererOUIdentifier:
+        Certificate: cacerts/ca.pem
+        OrganizationalUnitIdentifier: orderer' >"$CERT_DIR/msp/config.yaml"
+
+    fi
+
+    if [[ $TYPE == "peer" ]]; then
+        generate_peer_certs "$CERT_DIR" "$USER"
+    fi
+    if [[ $TYPE == "admin" ]]; then
+        generate_user_certs "$CERT_DIR" "$USER" "$TYPE"
+    fi
+
+    find . -name "*.csr" -print0 | xargs -0 rm
+
+}
+
+function orderer_cert() {
+    TYPE=$1 #orderer user
+    USER=$2 #orderer.example.com
+
+    mkdir -p organizations/ordererOrganizations/example.com/ca
+    mkdir -p organizations/ordererOrganizations/example.com/msp/cacerts
+    mkdir -p organizations/ordererOrganizations/example.com/msp/tlscacerts
+    mkdir -p organizations/ordererOrganizations/example.com/orderers
+    mkdir -p organizations/ordererOrganizations/example.com/tlsca
+
+    CERT_DIR=organizations/ordererOrganizations/example.com
+
+    if [ ! -f "$CERT_DIR/ca/ca-key.pem" ]; then
+
+        cfssl gencert -initca "${PWD}/organizations/cfssl/ca-orderer.json" | cfssljson -bare "$CERT_DIR/ca/ca"
+
+        cp "$CERT_DIR/ca/ca.pem" "$CERT_DIR/tlsca/tlsca.example.com-cert.pem"
+
+        cp "$CERT_DIR/ca/ca.pem" "$CERT_DIR/msp/cacerts/"
+        cp "$CERT_DIR/ca/ca.pem" "$CERT_DIR/msp/tlscacerts/"
```

Comment:
This needs to be called tlsca.example.com-cert.pem for the test-network instructions to work, e.g.

https://hyperledger-fabric.readthedocs.io/en/latest/test_network.html

`peer chaincode invoke -o localhost:7050 --ordererTLSHostnameOverride orderer.example.com --tls --cafile "${PWD}/organizations/ordererOrganizations/example.com/orderers/orderer.example.com/msp/tlscacerts/tlsca.example.com-cert.pem" -C mychannel -n basic --peerAddresses localhost:7051 --tlsRootCertFiles "${PWD}/organizations/peerOrganizations/org1.example.com/peers/peer0.org1.example.com/tls/ca.crt" --peerAddresses localhost:9051 --tlsRootCertFiles "${PWD}/organizations/peerOrganizations/org2.example.com/peers/peer0.org2.example.com/tls/ca.crt" -c '{"function":"InitLedger","Args":[]}'`